include/cuco/detail/static_map/kernels.cuh

```diff
@@ -76,33 +76,55 @@ CUCO_KERNEL __launch_bounds__(BlockSize) void insert_or_assign(InputIt first,
  *
  * @note Callable object to perform binary operation should be able to invoke as
  * Op(cuda::atomic_ref<T,Scope>, T>)
+ * @note If `HasInit` is `true` and if `init` value equals to the `sentinel value`, we directly
+ * `apply` the `op` instead of atomic store and then waiting for the payload to get materalized.
+ * This has potential speedups when sizeof(value_type) > 8.
```

Comment:
Is the only relevant part here the size of the value or is it the size of the pair, i.e., when using packed CAS vs. CAS + dependent write?

Reply:
I think its size of `value_type` of the **map**, because that will determine on which type we are performing CAS operation.

Reply:
Right. The exact size value might change in the future as we're about to get 128b CAS on Hopper+ exposed through libcu++. That means we can use the packed CAS approach with up-to 16B value types